### PR TITLE
Notification service: throwing errors on unimplemented base class methods

### PR DESCRIPTION
**Pull request by promaty**

### CODE REVIEW

File: packages/services/src/models/notification-scanners/NotificationScannerBaseModel.js [hunk 1]
```diff
@@ -16,19 +16,29 @@ export default class NotificationScannerBaseModel {
...
+   * return [
+   *   { 
+   *     address: null,
+   *     details: {
+   *       emailTemplateModel: {}
+   *     }
+   *   }
+   * ]
+   */
+  async scan() { throw 'subclass responsibility' }
```
Comment (sohkai) [inline packages/services/src/models/notification-scanners/NotificationScannerBaseModel.js:30]: This works, but it's generally better to wrap it in a `new Error()` so the stack trace and etc. get included.

Comment (facuspagnuolo) [reply packages/services/src/models/notification-scanners/NotificationScannerBaseModel.js:30]: Yup, let's always use `throw Error` which I think it's the shortcut for `throw new Error()`

Comment (facuspagnuolo) [reply packages/services/src/models/notification-scanners/NotificationScannerBaseModel.js:30]: Yup, let's always use `throw Error` which I think it's the shortcut for `throw new Error()`

Comment (promaty) [reply packages/services/src/models/notification-scanners/NotificationScannerBaseModel.js]: Added!

**Pull request merged**